config: Allow user-specified kernel params to take priority

jodh-intel opened the pull request:
Any `kernel_params=` values specified in the config file should take
priority over any in-built defaults (for better or for worse).

Fixes #963.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

devimc commented:
looks good, could you add a ```WARNING``` in the configuration file specifying that user kernel params take priority?

jodh-intel commented:
@devimc - good call! I've added a comment to the config file and also added an extra log call (which only does that string manipulation when in debug mode).

I'm also going to raise a PR on virtcontainers to add an equivalent `default-kernel-parameters` log call for the other kernel params set by vc's qemu code...

devimc commented:
lgtm

jodh-intel commented:
Here's the vc PR: https://github.com/containers/virtcontainers/pull/586/

jodh-intel commented:
With this PR and https://github.com/containers/virtcontainers/pull/586, the logs will have entries like this:

```
level=debug default-kernel-parameters="init=/usr/lib/systemd/systemd systemd.unit=kata-containers.target systemd.mask=systemd-networkd.service systemd.mask=systemd-networkd.socket ip=::::::35ea95e734533009eddde966e7040d230e90baad67baf71c3e057f9c220a3a20::off::" source=runtime                                  
level=debug default-kernel-parameters="root=/dev/pmem0p1 rootflags=dax,data=ordered,errors=remount-ro rw rootfstype=ext4 tsc=reliable no_timer_check rcupdate.rcu_expedited=1 i8042.direct=1 i8042.dumbkbd=1 i8042.nopnp=1 i8042.noaux=1 noreplace-smp reboot=k panic=1 console=hvc0 console=hvc1 initcall_debug iommu=off cryptomgr.notests net.ifnames=0 pci=lastbus=0 debug systemd.show_status=true systemd.log_level=debug" source=virtcontainers subsystem=qemu  
```

Note the `source=` and `subsystem=` fields.

jodh-intel commented:
Coveralls report: https://coveralls.io/jobs/33246934

Merging...
